src/Transaction.js

```diff
@@ -3,12 +3,38 @@
  * An account of what has happened.
  */
 
-export default function Transaction (type, payload, complete) {
-  return {
-    type     : `${ type }`,
-    error    : false,
-    active   : arguments.length > 1,
-    payload  : payload,
-    complete : complete
+import coroutine  from './coroutine'
+import flatten    from './flatten'
+import eventually from './eventually'
+
+export default function Transaction (action, payload) {
+  this.action  = action
+  this.type    = `${ action }`
+  this.active  = arguments.length > 1
+  this.payload = payload
+}
+
+Transaction.prototype = {
+  action   : null,
+  active   : false,
+  error    : false,
+  payload  : undefined,
+  complete : false,
+
+  execute(params, onNext, onComplete, scope) {
+    let body = this.action.apply(null, flatten(params))
+
+    return coroutine(body, (error, payload, done) => {
```

Comment:
Yes. Every time it yields a new value, state rolls forward with that new value. This is how we can do optimistic updates without having to clean up.